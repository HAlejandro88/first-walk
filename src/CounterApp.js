import React, { Fragment, useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({ value }) => {

    const [ counter, setCounter ] = useState(value); // Destructuro el hook de useState que viene el valor y funcion del Valor
    

    const handleAdd = e => setCounter(counter + 1);

    const handleSubstract = (e) => {
        setCounter( c => c - 1 );
    }
    const handleReset = e => setCounter(value);

    return (
        <Fragment>
            <h1>Counter App</h1>
            <h2>{ counter }</h2>
            <button onClick={ handleAdd }>+1</button>
            <button onClick={ handleReset }>Reset</button>
            <button onClick={ handleSubstract }>-1</button>
        </Fragment>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number
}


export default CounterApp;