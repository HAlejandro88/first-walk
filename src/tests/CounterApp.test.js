import React from 'react';
import CounterApp from '../CounterApp';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

describe('Prubas en el <CounterApp />', () => {
    let wrapper = shallow(<CounterApp />);

    beforeEach(() => {
         wrapper = shallow(<CounterApp />);
    })

    test('debe de mostrar e counter app correcatemnet', () => {
        expect(wrapper).toMatchSnapshot();

    });

    test('debe de mostrar el valor por defecto de 100', () => {
        const wrapper = shallow(<CounterApp value={100} />);
        const counterText =  wrapper.find('h2').text().trim();
        expect(counterText).toBe('100')
    });

    test('debe de incremetar en 1 con el boton +1', () => {
        const wrapper = shallow(<CounterApp value={10} />);
        wrapper.find('button').at(0).simulate('click');
        const counterText =  wrapper.find('h2').text().trim();
        expect(counterText).toBe('11');
    });

    test('debe de decrementar en 1 con el boton -1', () => {
        const wrapper = shallow(<CounterApp value={10} />);
        wrapper.find('button').at(2).simulate('click');
        const counterText =  wrapper.find('h2').text().trim();
        expect(counterText).toBe('9');
    });

    test('debe de colocar el valor por defecto el boton reset ', () => {
        const wrapper = shallow(<CounterApp value={105} />);

        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(1).simulate('click');

        const counterText =  wrapper.find('h2').text().trim();

        expect(counterText).toBe('105');
       
    });
})