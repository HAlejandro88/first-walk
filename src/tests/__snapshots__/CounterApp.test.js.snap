// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prubas en el <CounterApp /> debe de mostrar e counter app correcatemnet 1`] = `
<Fragment>
  <h1>
    Counter App
  </h1>
  <h2>
     
     
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
